ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/TableOptimizeItem.java

```diff
@@ -922,6 +928,27 @@ public void checkTaskExecuteTimeout() {
     }
   }
 
+  /**
+   * If task execute failed after max retry, clear tasks.
+   */
+  public void clearFailedTasks() {
+    tasksLock.lock();
+    try {
+      int maxRetry = optimizeMaxRetry();
+      Optional<OptimizeTaskItem> failedTask =
+          optimizeTasks.values().stream().findAny().filter(task -> task.getOptimizeRuntime().getRetry() > maxRetry);
```

Comment:
@hzluting 
- Should `findAny()` be placed at the end
- Add filter `OptimizeStatus.Failed == task.getOptimizeRuntime().getStatus()`

What do you think?

Reply:
Yes, thank you for your comment

Reply:
@zhongqishang I fix it in #1168